### PR TITLE
feat(): support Angular and rxjs 6

### PR DESCRIPTION
**Pull request by jamesdaniels**
### Checklist

   - Issue number for this PR: #1602, #1538
   - Docs included?: n/a 
   - Test units included?: yes
   - In a clean directory, `yarn install`, `yarn test` run successfully? yes

### Description

Adding rxjs and angular 6 support. ~Dropped database-deprecated, since there were a lot of changed needed there and it's no longer worth maintaining IMO.~

Huge shout out to @jymdman for doing the ground work in each module.

### CODE REVIEW

**Comment by jamesdaniels** [comment]
Closing @kawamoto's pull request in favor of this one; I pulled in some of his changes so gave him co-author status. He's signed the CLA as verified in #1582

**Comment by davideast** [comment]
@jamesdaniels @robertbaker It's not being favored. James has been working on various AngularFire issues and we haven't communicated with each other on this one due to Google I/O.

@jamesdaniels I'm closing this PR because other community members were in the middle of working on this. It also contains the deletion of the `database-deprecated` module which is something we need clearer notice on before handling.

**Comment by GaryB432** [comment]
Can the status of angular and rxjs 6 be clarified please?

**Comment by davideast** [comment]
@GaryB432 We are clarifying who is working the PR in #1582. Who ever takes the issue first will work on it. We will also check on the status to make sure it is implemented in a timely manner. In the meantime please use the compat library until we get fully up to speed.

**Comment by davideast** [comment]
@robertbaker I just noticed that @jymdman sent in the PRs which were small and done by feature. This PR looks large because it includes the removal of `database-deprecated`. The RxJS changes are small. However, the `database-deprecated` removal should be done in another PR.

**Comment by jamesdaniels** [comment]
@robertbaker sorry for making this look "preferred". I already merged in several of @jymdman's PRs which brought the modules up to speed with pipeable operators. For the most part here I just pulled in the pieces of @kawamoto's pull request that were still needed; largely just updating the paths, build, karma.conf, and getting the tests green and please note I gave him co-contributor status here. I did this simply so I could move faster in merging this work; not to squash any community effort.

@davideast I'll do the work and send another PR which gets database-deprecated up to speed with Firebase JS SDK 5.0 and pipeable operators, update this to get it up to speed with 6, so we can merge this--then put a proposal up for removing database-deprecated.

**Comment by jamesdaniels** [comment]
@robertbaker @GaryB432 and yes, we're sorry for the lack of communication here; both @davideast and I have been very busy prepping for, speaking at, and staffing Google I/O. I'm just trying to jump on this now that we're in the clear. I want everything fixed up so I can cut a new RC Monday AM.

**Comment by jamesdaniels** [comment]
Released as `angularfire2@next`

**Comment by jamesdaniels** [comment]
Wow, googlebot does not handle branches with co-contributors well at all.

File: test/ng-build/index.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ function packageAngularFire() {
   const res = spawnSync('sh', ['pack.sh']);
   console.log(`------------ FINISHED PACKAGING VERSION ${PACKAGED_VERSION} ------------`);
   console.log(`------------ INSTALLING VERSION ${PACKAGED_VERSION} ------------`);
-  if (shell.exec(`cd ng5 && npm i firebase ../${PACKAGED_VERSION}`).code !== 0) {
+  if (shell.exec(`cd ng5 && yarn i firebase ../${PACKAGED_VERSION}`).code !== 0) {
```
Comment (sarunint) [inline test/ng-build/index.js]: `yarn add`?

Comment (jamesdaniels) [reply test/ng-build/index.js]: good catch

**Comment by maguro** [comment]
Wow, this **_mega change_** gets pushed out in a release candidate?!?!

**Comment by davideast** [comment]
@maguro AngularFire is a combination of Firebase, Angular, and RxJS. Within a short period of time all libraries had major upgrades with breaking changes. We were already in the middle of a release candidate and were forced to make the changes. We opted to update to the major releases for everyone before going final. If we went final before upgrading then we would have had to break immediately.

**Pull request merged**